prow/cmd/config-bootstrapper/main.go

```diff
@@ -127,25 +102,52 @@ func main() {
 		return nil
 	})
 
-	for cm, data := range updateconfig.FilterChanges(pluginAgent.Config().ConfigUpdater, changes, logrus.NewEntry(logrus.StandardLogger())) {
-		if cm.Namespace == "" {
-			cm.Namespace = configAgent.Config().ProwJobNamespace
-		}
+	for cm, data := range updateconfig.FilterChanges(configUpdater, changes, defaultNamespace, logrus.NewEntry(logrus.StandardLogger())) {
 		logger := logrus.WithFields(logrus.Fields{"configmap": map[string]string{"name": cm.Name, "namespace": cm.Namespace, "cluster": cm.Cluster}})
 		configMapClient, err := updateconfig.GetConfigMapClient(client.CoreV1(), cm.Namespace, buildClusterCoreV1Clients, cm.Cluster)
 		if err != nil {
 			logrus.WithError(err).Errorf("Failed to find configMap client")
 			continue
 		}
-		if err := updateconfig.Update(&updateconfig.OSFileGetter{Root: o.sourcePath}, configMapClient, cm.Name, cm.Namespace, data, bootstrapMode, nil, logger); err != nil {
+		if err := updateconfig.Update(&updateconfig.OSFileGetter{Root: sourcePath}, configMapClient, cm.Name, cm.Namespace, data, bootstrapMode, nil, logger); err != nil {
 			logger.WithError(err).Error("failed to update config on cluster")
 			errors++
 		} else {
 			logger.Info("Successfully processed configmap")
 		}
 	}
+	return errors
+}
+
+func main() {
+	logrusutil.ComponentInit("config-bootstrapper")
+
+	o := gatherOptions()
+	if err := o.Validate(); err != nil {
+		logrus.WithError(err).Fatal("Invalid options")
+	}
+
+	configAgent := &config.Agent{}
+	if err := configAgent.Start(o.configPath, o.jobConfigPath); err != nil {
+		logrus.WithError(err).Fatal("Error starting config agent.")
+	}
+
+	pluginAgent := &plugins.ConfigAgent{}
+	if err := pluginAgent.Start(o.pluginConfig, true); err != nil {
+		logrus.WithError(err).Fatal("Error starting plugin configuration agent.")
+	}
+
+	client, err := o.kubernetes.InfrastructureClusterClient(o.dryRun)
+	if err != nil {
+		logrus.WithError(err).Fatal("Error getting Kubernetes client.")
+	}
+
+	buildClusterCoreV1Clients, err := o.kubernetes.BuildClusterCoreV1Clients(o.dryRun)
+	if err != nil {
+		logrus.WithError(err).Fatal("Error getting Kubernetes clients for build cluster.")
+	}
 
-	if errors > 0 {
+	if errors := run(o.sourcePath, configAgent.Config().ProwJobNamespace, pluginAgent.Config().ConfigUpdater, client, buildClusterCoreV1Clients); errors > 0 {
 		logrus.WithField("fail-count", errors).Fatalf("errors occurred during update")
```

Comment:
I think you want `len(errors)` for the fail count?

Reply:
`errors` is int, right?
I did not change the type.